cfpb/cfgov-refresh
# Performance fixes --- preloading assets and eliminating script tags

Performance fixes --- preloading assets and eliminating script tags

## Changes

- Modified `cfgov/jinja2/v1/_layouts/base-common.html` to remove unnecessary script tags and combine them where possible.

- Modified `cfgov/jinja2/v1/_layouts/base.html` to preload web fonts, CSS, and JS. This is largely based on the following articles:
   - https://www.keycdn.com/blog/resource-hints/
   - https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf
   - https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/webfont-optimization

- Modified `cfgov/jinja2/v1/index.html` to preload images on the homepage.

## Testing

1. Visit `http://localhost:8000/` 
2. Click on the performance tab in Chrome dev tools.
3. Click the reload button, while ensuring the `Screenshots` checkbox is enabled. 
4. Open `base.html` and remove all code contained in the `preload` block.
5. Revisit the Chrome dev tools and hit the reload button again. 
6. Compare the results by selecting the dropdown in the tool bar (hint: it should say `localhost #2`)
7. Verify visually that the render time is faster in the first performance test.
8. View the network panel and Chrome and verify the order is: HTML, CSS, Web fonts, common.js images, on-demand js, and then analytics related code.

## Screenshots
<img width="554" alt="screen shot 2018-03-18 at 11 09 29 am" src="https://user-images.githubusercontent.com/1696212/37567542-dc163d40-2a9e-11e8-81d7-bb3349f1624d.png">

`locahost #1` has the fastest render time, and has the preload / dns-lookup code enabled. `locahost #2` and `locahost #3` are variations of our current setup.

## Notes

- I expect the results to be much better when running with Http2, Gzip, and Akamai.

## Todos

-

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
